### PR TITLE
Add a 'leave page' button to the vasu editor and preview pages

### PR DESCRIPTION
**Pull request by zch**
#### Summary
This is in order to improve the UX of this flow. The action is the same
as pressing the back button in the browser.

### CODE REVIEW

File: frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx [hunk 1]
```diff
@@ -145,12 +146,18 @@ export default React.memo(function VasuEditPage({
...
+              <Button
+                text={i18n.vasu.checkInPreview}
+                disabled={status.state != 'clean'}
+                onClick={() => history.push(`/vasu/${vasu.id}`)}
+                primary
+              />
+              <Button
+                text={i18n.vasu.leavePage}
+                onClick={() => history.goBack()}
+              />
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx:159]: "Poistu" is a little bit confusing term here.

1. Open vasu from child page
2. Click Edit
3. Click check in preview
4. Click Edit
5. Click Leave page 4 times to actually leave the page.

File: frontend/src/lib-customizations/espoo/employee/assets/i18n/fi.ts [hunk 2]
```diff
@@ -894,6 +894,7 @@ export const fi = {
     lastPublished: 'Viimeksi julkaistu huoltajalle',
     vasuDiscussion: 'Vasukeskustelu',
     evaluationDiscussion: 'Arviointikeskustelu',
+    leavePage: 'Poistu',
```
[the remark this answers is not in the document]
Comment (zch) [reply frontend/src/lib-customizations/espoo/employee/assets/i18n/fi.ts:897]: Discussed with UX. The button is in the correct container, but we left-aligned it to create a clear separation between the different actions.

File: frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx (hunk 1 above)
Comment (zch) [reply frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx]: Changed the target of the button to always go back to the child information page.

File: frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx [hunk 3]
```diff
@@ -151,12 +156,17 @@ export default React.memo(function VasuEditPage({
...
-            />
+            <FullWidthDiv>
+              <ButtonContainer>
+                <Button
+                  text={i18n.vasu.checkInPreview}
+                  disabled={status.state != 'clean'}
+                  onClick={() => history.push(`/vasu/${vasu.id}`)}
+                  primary
+                />
+                <LeaveVasuPageButton childId={vasu.basics.child.id} />
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx:167]: Should this be disabled too if state is not clean?

File: frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx [hunk 4]
```diff
@@ -151,12 +156,17 @@ export default React.memo(function VasuEditPage({
             {showSpinner && <Spinner />}
           </StatusContainer>
           {vasu && (
-            <Button
-              text={i18n.vasu.checkInPreview}
-              disabled={status.state != 'clean'}
-              onClick={() => history.push(`/vasu/${vasu.id}`)}
-              primary
-            />
+            <FullWidthDiv>
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx:159]: The button position flickers when StatusContainer changes (textual changes and spinner). Oh well, if this is what the UX department wanted, then let's roll with it 😃

File: frontend/src/employee-frontend/components/vasu/components/LeaveVasuPageButton.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  )
+})
+
+const ExitButtonWrapper = styled.div`
+  flex: 1;
+`
+
+export const FullWidthDiv = styled.div`
+  width: 100%;
+`
```
Comment (kimifoud) [inline frontend/src/employee-frontend/components/vasu/components/LeaveVasuPageButton.tsx:36]: `FullWidthDiv` is not used by `LeaveVasuPageButton` so it should probably be defined somewhere else.

File: frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx (hunk 3 above)
Comment (zch) [reply frontend/src/employee-frontend/components/vasu/VasuEditPage.tsx]: Good catch! Fixed.

File: frontend/src/employee-frontend/components/vasu/components/LeaveVasuPageButton.tsx (hunk 5 above)
Comment (zch) [reply frontend/src/employee-frontend/components/vasu/components/LeaveVasuPageButton.tsx]: Moved to lib-components/layouts

**Review by kimifoud — APPROVED** [review]

**Pull request merged**